Testing : Code coverage report, reintegrate commit from tswfi, and change internal linking (#1684)
PrestaShop/docs: testing/unit-tests/how-to-execute-tests.md

{{% notice tip %}}
Hint: `--debug` without `--filter` gives a nice list to filter with.
{{% /notice %}}

## Checking for code coverage in phpunit tests

> To check if you covered everything in your tests cases its best to run the tests with [phpunit coverage report](https://phpunit.readthedocs.io/en/9.5/code-coverage-analysis.html).

Reviewer: ```suggestion
To check if you covered everything in your test cases, it's best to run the tests with [phpunit coverage report](https://phpunit.readthedocs.io/en/9.5/code-coverage-analysis.html).
```

> First get your [environment up and running]({{< relref "/8/basics/installation" >}}). 

Reviewer: ```suggestion
First, get your [environment up and running]({{< relref "/8/basics/installation" >}}). 
```

Then, [install and enable Xdebug](https://xdebug.org/docs/install).

Finally, run your tests with coverage enabled:

```bash
XDEBUG_MODE=coverage vendor/bin/phpunit --coverage-text -c tests/Unit/phpunit.xml tests/Unit/PrestaShopBundle/Command/ConfigCommandTest.php
```

{{% notice note %}}
You may use a [Dockerized environment to run the project]({{< relref "/8/basics/installation/environments/docker" >}}).

> After that is up and running you need to compile and enable xdebug:

Reviewer: ```suggestion
After that is up and running, you need to compile and enable Xdebug:
```

```bash
docker compose exec prestashop_container pecl install xdebug #prestashop_container is the container's name
docker compose exec prestashop_container docker-php-ext-enable xdebug #prestashop_container is the container's name
```

Then run your tests with coverage enabled:

```bash
docker compose exec -e XDEBUG_MODE=coverage prestashop_container vendor/bin/phpunit --coverage-text -c tests/Unit/phpunit.xml tests/Unit/PrestaShopBundle/Command/ConfigCommandTest.php
```
{{% /notice %}}

> This will give you a nice report of how much of your code was covered by the tests and you can extend your tests to get closer to a perfect score of 100%.

Reviewer: ```suggestion
This will give you a nice report of how much of your code was covered by the tests, and you can extend your tests to get closer to a perfect score of 100%.
```
